test/conftest.py

```diff
@@ -7,20 +7,7 @@ def pytest_addoption(parser) -> None:
     parser.addoption(
         "--run-emulation", action="store_true", default=False, help="run emulation tests"
     )
-
-
-def pytest_configure(config):
-    config.addinivalue_line("markers", "emulation: mark test requiring qemu binfmt_misc to run")
-
-
-def pytest_collection_modifyitems(config, items) -> None:
-    if config.getoption("--run-emulation"):
-        # --run-emulation given in cli: do not skip emulation tests
-        return
-    skip_emulation = pytest.mark.skip(reason="need --run-emulation option to run")
-    for item in items:
-        if "emulation" in item.keywords:
-            item.add_marker(skip_emulation)
```

Comment:
I'm not sure it's better - the markers would allow a user to statically see all the tests marked with emulation, and read in the normal pytest style, where a test is decorated with it's properties. This is now just procedural manual hacks in each test rather than declarative markers, and you have to look inside for skips, and pytest doesn't know beforehand it's skippable.

The marker should have been in pyproject.toml and not in contest.py, but handing markers in this way was fine in pytest, IMO. There's actually at least one plugin for "opt-in" markers (https://github.com/reece/pytest-optional-tests)  that IMO would have been even better - it's not the best maintained, though; I've manually implemented it with fixes inside plumbum at https://github.com/tomerfiliba/plumbum/blob/master/tests/conftest.py, for example. I'm not sure if @reece would be interested in having some help maintaining the plugin.